spec/API_specification/array_api/data_type_functions.py

```diff
@@ -117,6 +117,36 @@ def iinfo(type: Union[dtype, array], /) -> iinfo_object:
           integer data type.
     """
 
+def is_type(dtype: dtype, kind: Union[dtype, str, Tuple[Union[dtype, str], ...]]) -> bool:
```

Comment:
should be `is_dtype`?

Reply:
Discussed in the OP. In short, if we use `is_dtype`, we have three different naming conventions:

```
astype
result_type
is_dtype
```

Chose `is_type` to be consistent with `result_type`. Discussed in consortium meeting. If consistency is valued, another option is `istype`.

Reply:
Yeah, this is annoying.  I am considering bikeshedding it also on NumPy, maybe someone has a good idea.
`type` is weird, and `result_type` and `astype` should be more reasonably `asdtype` and especially `result_dtype` (for `astype`, somehow I find it a bit less weird).

So I much prefer `dtype`, but I am a bit on the hedge of whether aiming for consistency is better here.  There was also discussion about `isdtype/istype` vs. with underscore.  Without underscore does make some sense also w.r.t. `isinstance` or `isclass` maybe @honno's rule of is and as don't need underscore is right.

Reply:
@seberg I share your sentiments, discussing in NumPy sounds like a good idea.

Regarding consistency: I'd argue that `astype` and `result_type` are the outliers; we already have `.dtype` and consistently use "dtype" in language everywhere. "type" has a very different meaning in Python.

Reply:
There are many similar things in Python which are exceptions, like `isinstance`, `hasattr`, etc.  I am not sure if Python considers these badly named.  That said, I don't have a strong opinion either way.
I do agree that `dtype` is better and `result_type` is badly named (I really prefer `common_dtype`, numpy has `common_type`, but that needs to go anyway).

Reply:
Re PEP8 and last comments I agree with @seberg. Sebastian, was it discussed in NumPy (sorry, I missed most of the last community calls)?

Reply:
Renamed to `isdtype`. In general, unless a function is a contracted name (e.g., `vecdot` or `matmul`, etc), the specification prefers using underscores to separate distinct words. The only functions in the specification which do not include underscores separating two words are:

- `asarray`
- `astype`
- `isfinite`
- `isinf`
- `isnan`

Given the precedent of `isfinite`, `isinf`, and `isnan`, I've updated this PR to use `isdtype`, preferring consistency over PEP8 purity.